From 48191396215662084386276aa7a7a48cfb34e249 Mon Sep 17 00:00:00 2001
From: Daniel Jarai <jaraidaniel@gmail.com>
Date: Sun, 2 Sep 2018 00:34:14 +0200
Subject: [PATCH 01/14] msm8974-common: Make the external camera provider
 ignore internal cameras

* The external camera provider occupies our camera v4l2 nodes,
  potentially crashing the kernel driver and blocking the camera HAL.
  Unfortunately, there's no easy way to just disable it, so let's just
  tell it to ignore the internal video devices.

Change-Id: I00c67de1b94a1f7eca1bf458d1d0e4de3507aa81
Signed-off-by: Daniel Jarai <jaraidaniel@gmail.com>
---
 configs/external_camera_config.xml | 15 +++++++++++++++
 msm8974.mk                         |  3 +++
 2 files changed, 18 insertions(+)
 create mode 100644 configs/external_camera_config.xml

diff --git a/configs/external_camera_config.xml b/configs/external_camera_config.xml
new file mode 100644
index 0000000..0f907f0
--- /dev/null
+++ b/configs/external_camera_config.xml
@@ -0,0 +1,15 @@
+<ExternalCamera>
+    <Provider>
+        <!-- Internal video devices to be ignored by external camera HAL -->
+        <ignore>
+            <id>0</id>
+            <id>1</id>
+            <id>2</id>
+            <id>3</id>
+            <id>32</id>
+            <id>33</id>
+            <id>38</id>
+            <id>39</id>
+        </ignore>
+    </Provider>
+</ExternalCamera>
diff --git a/msm8974.mk b/msm8974.mk
index 96a7f45..ab8d106 100644
--- a/msm8974.mk
+++ b/msm8974.mk
@@ -105,6 +105,9 @@ PRODUCT_PACKAGES += \
     libxml2 \
     Snap
 
+PRODUCT_COPY_FILES += \
+    $(LOCAL_PATH)/configs/external_camera_config.xml:$(TARGET_COPY_OUT_VENDOR)/etc/external_camera_config.xml
+
 # DRM
 PRODUCT_PACKAGES += \
     android.hardware.drm@1.0-impl \
-- 
2.17.1

